custom_components/mikrotik_router/mikrotik_controller.py

```diff
@@ -1215,17 +1233,34 @@ def get_system_resource(self):
         )
 
         tmp_uptime = 0
-        tmp = re.split(r"(\d+)[h]", self.data["resource"]["uptime_str"])
+        tmp = re.split(r"(\d+)[s]", self.data["resource"]["uptime_str"])
         if len(tmp) > 1:
             tmp_uptime += int(tmp[1])
+        tmp = re.split(r"(\d+)[m]", self.data["resource"]["uptime_str"])
+        if len(tmp) > 1:
+            tmp_uptime += int(tmp[1]) * 60
+        tmp = re.split(r"(\d+)[h]", self.data["resource"]["uptime_str"])
+        if len(tmp) > 1:
+            tmp_uptime += int(tmp[1]) * 3600
         tmp = re.split(r"(\d+)[d]", self.data["resource"]["uptime_str"])
         if len(tmp) > 1:
-            tmp_uptime += int(tmp[1]) * 24
+            tmp_uptime += int(tmp[1]) * 86400
         tmp = re.split(r"(\d+)[w]", self.data["resource"]["uptime_str"])
         if len(tmp) > 1:
-            tmp_uptime += int(tmp[1]) * 24 * 7
+            tmp_uptime += int(tmp[1]) * 604800
+
+        now = datetime.now().replace(microsecond=0)
+        uptime_tm = datetime.timestamp(now - timedelta(seconds=tmp_uptime))
+        update_uptime = False;
+        if self.data["resource"]["uptime"] is None or self.data["resource"]["uptime"] == 0:
+            update_uptime = True;
+        else:
+            uptime_old = datetime.timestamp(datetime.fromisoformat(self.data["resource"]["uptime"]))
+            if uptime_tm > uptime_old:
```

Comment:
Something like this should solve any possible issues with shifting imo:
`if uptime_tm > uptime_old + 10:`

Reply:
Totally agree. A 10 seconds of margin should solve possible issues with shifting imo